components/Msig/AddSigner.tsx

```diff
@@ -0,0 +1,124 @@
+  pendingMsgContext
+}: AddSignerProps) => {
+  const router = useRouter()
+  const wallet = useWallet()
+  // @ts-expect-error
+  const { serializeParams } = useWasm()
+  const { Address, AvailableBalance, NumApprovalsThreshold } = useMsig()
+
+  // Input states
+  const [signer, setSigner] = useState<string>('')
+  const [increase, setIncrease] = useState<boolean>(false)
+
+  // Transaction states
+  const [txState, setTxState] = useState<TxState>(TxState.FillingForm)
+  const [txFee, setTxFee] = useState<FilecoinNumber | null>(null)
+
+  // Create message from input
+  const message = useMemo<Message | null>(
+    () =>
+      // Manually check signer validity to prevent passing invalid addresses to serializeParams.
+      // This can happen due to multiple rerenders when using setIsValid from InputV2.Address.
+      validateAddressString(signer)
+        ? new Message({
+            to: Address,
+            from: wallet.address,
+            nonce: 0,
+            value: 0,
+            method: MsigMethod.PROPOSE,
+            params: Buffer.from(
+              serializeParams({
+                to: Address,
+                value: '0',
+                method: MsigMethod.ADD_SIGNER,
+                params: Buffer.from(
+                  serializeParams({
+                    signer,
+                    increase
+                  }),
+                  'hex'
+                ).toString('base64')
+              }),
+              'hex'
+            ).toString('base64'),
+            gasPremium: 0,
+            gasFeeCap: 0,
+            gasLimit: 0
+          })
+        : null,
+    [signer, increase, Address, wallet.address, serializeParams]
+  )
+
+  return (
+    <Transaction.Form
+      title='Add a signer'
+      description='Please enter the new signer address below'
+      warning="You're about to add another signer to your Safe. Please make sure you know and trust the new owner as they will be able to withdraw funds from your Safe."
+      msig
+      method={MsigMethod.ADD_SIGNER}
+      message={message}
+      total={txFee}
+      txState={txState}
+      setTxState={setTxState}
+      maxFee={wallet.balance}
+      txFee={txFee}
+      setTxFee={setTxFee}
+      onComplete={() => navigate(router, { pageUrl: PAGE.MSIG_HISTORY })}
+      walletProviderOpts={walletProviderOpts}
+      pendingMsgContext={pendingMsgContext}
+    >
+      <Transaction.Balance
+        address={wallet.address}
+        balance={wallet.balance}
+        msigBalance={AvailableBalance}
+      />
+      <InputV2.Address
+        label='New Signer'
+        autoFocus
+        value={signer}
+        onChange={setSigner}
+        disabled={txState !== TxState.FillingForm}
+      />
+      <InputV2.Toggle
+        label='Increase required approvals'
+        info={`From ${NumApprovalsThreshold} to ${NumApprovalsThreshold + 1}`}
+        checked={increase}
+        onChange={setIncrease}
+        disabled={txState !== TxState.FillingForm}
+      />
```

Comment:
When `txState > TxState.FillingForm`, I think this should show the actual # selected rather than the select component, and then hide the `info` prop